lib/Jsrt/Jsrt.cpp

```diff
@@ -1367,102 +1367,199 @@ CHAKRA_API JsPreventExtension(_In_ JsValueRef object)
     });
 }
 
+static JsErrorCode InternalGetPropertyRecord(Js::ScriptContext * scriptContext, JsValueRef key, _Out_ const Js::PropertyRecord ** propertyRecord)
+{
+    Assert(propertyRecord != nullptr);
+    *propertyRecord = nullptr;
+
+    if (VirtualTableInfo<Js::LiteralStringWithPropertyStringPtr>::HasVirtualTable(key))
+    {
+        Js::LiteralStringWithPropertyStringPtr * ptrStr = (Js::LiteralStringWithPropertyStringPtr*)key;
+        *propertyRecord = ptrStr->GetPropertyStringLazy()->GetPropertyRecord();
+    }
+    else if (VirtualTableInfo<Js::PropertyString>::HasVirtualTable(key))
+    {
+        Js::PropertyString * ptrStr = (Js::PropertyString*)key;
+        *propertyRecord = ptrStr->GetPropertyRecord();
+    }
+    else if (Js::JavascriptString::Is(key))
+    {
+        Js::JavascriptString * ptrStr = Js::JavascriptString::FromVar(key);
+        scriptContext->GetOrAddPropertyRecord(ptrStr->GetSz(), static_cast<int>(ptrStr->GetLength()),
+            (Js::PropertyRecord const **)propertyRecord);
+    }
+    else
+    {
+        return JsErrorInvalidArgument;
+    }
+
+    return JsNoError;
+}
+
+static JsErrorCode JsGetPropertyCommon(Js::ScriptContext * scriptContext, _In_ Js::RecyclableObject * object,
+    _In_ const Js::PropertyRecord * propertyRecord, _Out_ JsValueRef *value,
+    TTDRecorder& _actionEntryPopper)
+{
+    AssertMsg(scriptContext->GetThreadContext()->IsScriptActive(), "Caller is expected to be under ContextAPIWrapper!");
+    PERFORM_JSRT_TTD_RECORD_ACTION(scriptContext, RecordJsRTGetProperty, propertyRecord, object);
+
+    *value = Js::JavascriptOperators::GetPropertyNoCache(object, propertyRecord->GetPropertyId(), scriptContext);
```

Comment:
Why `GetPropertyNoCache`?

Reply:
We know the type and no cache info ptr. So, no waste time checking and cache related stuff.